TMGToolbox/input_output/importNetwork.py

```diff
@@ -356,12 +356,50 @@ def createCentroidConfiguration(name, listOfCentroidIds):
     folder.append(centroidConfig)
     return centroidConfig
 
+# Reads the modes file and defines all possible modes on the netowrk
+def defineModes(filename):
+    # Delete the default modes
+    sectionType = model.getType("GKVehicle")
+    for types in model.getCatalog().getUsedSubTypesFromType( sectionType ):
+        for s in iter(types.values()):
+            if s != None:
+                cmd = s.getDelCmd()
+                model.getCommander().addCommand(cmd)
+    model.getCommander().addCommand(None)
+    # make list of nodes and vehicles created
+    modes = []
+    vehicleTypes = []
+    # read the file
+    with open(filename, 'r') as f:
+        lines = f.readlines()
+    for line in lines:
+        if line[0] == 'a':
```

Comment:
You should check to make sure the line isn't blank here before testing if we are adding a new mode.